### PR TITLE
Filter values refactor

### LINKED ISSUE #1054: Refactor Filter-Values
Deleting all of the filter-values that inherit from base-interval-component.
Refactor invalid-value, multi-value-input, and null-input to typescript and change to request-v2 for tests.

### PR DESCRIPTION
**Pull request by holderb5**
Resolves #1054 

## License

I confirm that this contribution is made under an MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/reports/tests/integration/components/filter-values/value-input-test.ts [hunk 1]
```diff
@@ -4,12 +4,24 @@ import $ from 'jquery';
...
+}
 module('Integration | Component | filter values/value input', function(hooks) {
   setupRenderingTest(hooks);
+  setupMirage(hooks);
 
-  hooks.beforeEach(async function() {
-    this.filter = { values: arr([1000]) };
+  hooks.beforeEach(async function(this: TestContext) {
+    const fragmentFactory = this.owner.lookup('service:fragment-factory') as FragmentFactory;
+    this.filter = fragmentFactory.createFilter('metric', 'bardOne', 'adClicks', {}, 'bet', [1000]);
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/filter-values/value-input-test.ts:24]: 'bet' operator doesn't really make sense
```suggestion
    this.filter = fragmentFactory.createFilter('metric', 'bardOne', 'adClicks', {}, 'gt', [1000]);
```

File: packages/reports/addon/components/filter-values/null-input.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+ * Copyright 2020, Yahoo Holdings Inc.
+ * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
+ */
+import Component from '@glimmer/component';
+import Args from './args-interface';
+
+export default class NullInput extends Component<Args> {
+  /**
+   * @property {String} tagName
+   */
```
Comment (kevinhinterlong) [inline packages/reports/addon/components/filter-values/null-input.ts:11]:
```suggestion
```

File: packages/reports/addon/components/filter-values/multi-value-input.ts [hunk 3]
```diff
@@ -31,12 +39,12 @@ export default class MultiValueInput extends Component {
    * @param {String} tag - Add single value to the filter values list
```
Comment (kevinhinterlong) [inline packages/reports/addon/components/filter-values/multi-value-input.ts:39]: Remove extra type info from docs
```suggestion
   * @param tag - Add single value to the filter values list
```

File: packages/reports/addon/components/filter-values/multi-value-input.ts [hunk 4]
```diff
@@ -8,21 +8,29 @@
...
+   */
+  tags: (string | number)[] = [];
 
-@templateLayout(layout)
-@tagName('')
-export default class MultiValueInput extends Component {
   /**
    * @method init
    * @override
    */
```
Comment (kevinhinterlong) [inline packages/reports/addon/components/filter-values/multi-value-input.ts:29]:
```suggestion
```

File: packages/reports/addon/components/filter-values/dimension-select.ts [hunk 5]
```diff
@@ -58,7 +58,10 @@ export default class DimensionSelectComponent extends Component<DimensionSelectC
   get selectedDimensions() {
```
Comment (kevinhinterlong) [inline packages/reports/addon/components/filter-values/dimension-select.ts:58]: Can we explicitly type this
```suggestion
  get selectedDimensions(): NaviDimensionModel[] {
```

File: packages/reports/addon/components/filter-values/multi-value-input.ts [hunk 6]
```diff
@@ -8,21 +8,29 @@
...
+
+interface MultiValueInputArgs {
+  filter: FilterFragment;
+  onUpdateFilter(changeSet: Partial<FilterFragment>): void;
+}
+
+export default class MultiValueInput extends Component<MultiValueInputArgs> {
+  /**
+   * @property {Array} tags
+   */
```
Comment (kevinhinterlong) [inline packages/reports/addon/components/filter-values/multi-value-input.ts:23]:
```suggestion
```

File: packages/reports/tests/integration/components/filter-values/invalid-value-test.ts [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+import { TestContext as Context } from 'ember-test-helpers';
+import FilterFragment from 'navi-core/models/bard-request-v2/fragments/filter';
+import FragmentFactory from 'navi-core/services/fragment-factory';
+//@ts-ignore
+import { setupMirage } from 'ember-cli-mirage/test-support';
+
+interface TestContext extends Context {
+  fragmentFactory: FragmentFactory;
+  filter: FilterFragment;
+}
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/filter-values/invalid-value-test.ts:14]: Update component integration tests to grab args from the component

File: packages/reports/addon/components/filter-values/multi-value-input.ts [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+import { set, action } from '@ember/object';
+import FilterFragment from 'navi-core/models/bard-request-v2/fragments/filter';
+
+interface MultiValueInputArgs {
+  filter: FilterFragment;
+  onUpdateFilter(changeSet: Partial<FilterFragment>): void;
+}
+
+export default class MultiValueInput extends Component<MultiValueInputArgs> {
+  tags: (string | number)[] = [];
```
[the remark this answers is not in the document]
Comment (holderb5) [reply packages/reports/addon/components/filter-values/multi-value-input.ts:21]: like `@tracked tags = [];`?

File: packages/reports/tests/integration/components/filter-values/range-input-test.ts [hunk 9]
```diff
@@ -9,10 +8,10 @@ import FragmentFactory from 'navi-core/addon/services/fragment-factory';
 //@ts-ignore
 import { setupMirage } from 'ember-cli-mirage/test-support';
```
Comment (kevinhinterlong) [inline packages/reports/tests/integration/components/filter-values/range-input-test.ts:10]:
```suggestion
```

**Review by kevinhinterlong — APPROVED** [review]

File: packages/reports/addon/components/filter-values/multi-value-input.ts [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+ */
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import FilterFragment from 'navi-core/models/bard-request-v2/fragments/filter';
+import { tracked } from '@glimmer/tracking';
+
+interface MultiValueInputArgs {
+  filter: FilterFragment;
+  onUpdateFilter(changeSet: Partial<FilterFragment>): void;
+}
```
Comment (jkusa) [inline packages/reports/addon/components/filter-values/multi-value-input.ts:19]: Can we use ` args-interface.ts` instead?

File: packages/reports/tests/integration/components/filter-values/invalid-value-test.ts [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+type ComponentArgs = InvalidValue['args'];
+interface TestContext extends Context, ComponentArgs {}
+
+module('Integration | Component | filter values/invalid value', function(hooks) {
+  setupRenderingTest(hooks);
+  setupMirage(hooks);
+
+  test('it renders', async function(this: TestContext, assert) {
+    assert.expect(1);
+
```
Comment (jkusa) [inline packages/reports/tests/integration/components/filter-values/invalid-value-test.ts:21]:
```suggestion
```

File: packages/reports/tests/integration/components/filter-values/invalid-value-test.ts [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+    this.fragmentFactory = this.owner.lookup('service:fragment-factory') as FragmentFactory;
+    this.filter = this.fragmentFactory.createFilter('dimension', 'bardOne', 'age', { field: 'id' }, 'in', []);
+    this.set('filter.values', undefined);
+    await render(hbs`<FilterValues::InvalidValue @filter={{this.filter}} />`);
+
+    assert.dom('.filter-values--selected-error').exists();
```
Comment (jkusa) [inline packages/reports/tests/integration/components/filter-values/invalid-value-test.ts:27]: Can we add an assert msg?

File: packages/reports/addon/components/filter-values/multi-value-input.ts (hunk 10 above)
Comment (holderb5) [reply packages/reports/addon/components/filter-values/multi-value-input.ts:19]: Can you explain this further?

Comment (jkusa) [reply packages/reports/addon/components/filter-values/multi-value-input.ts:19]:
```ts
import Args from './args-interface';
```

Comment (holderb5) [reply packages/reports/addon/components/filter-values/multi-value-input.ts:19]: and then would I do interface MultiValueInputArgs extends Args?

Comment (jkusa) [reply packages/reports/addon/components/filter-values/multi-value-input.ts:19]: Similar to what was done here: https://github.com/yahoo/navi/pull/1057/files#diff-87c52ce7e472e1f711c579c2a929aebfR8

**Review by jkusa — APPROVED** [review]

File: packages/reports/tests/integration/components/filter-values/multi-value-input-test.ts [hunk 13]
```diff
@@ -0,0 +1,85 @@
...
+  });
+
+  test('it renders', function(this: TestContext, assert) {
+    assert.expect(1);
+
+    assert.equal(
+      $('.emberTagInput-tag')[0].innerText.trim(),
+      this.filter.values[0],
+      'The value select contains an input with the first filter value as a tag'
+    );
```
Comment (mattgraham1995) [inline packages/reports/tests/integration/components/filter-values/multi-value-input-test.ts:39]:
```suggestion
    assert.dom('.emberTagInput-tag').hasText(
      this.filter.values[0],
      'The value select contains an input with the first filter value as a tag'
    );
```
Can we cleanup the jquery usage in this test?

File: packages/reports/tests/integration/components/filter-values/multi-value-input-test.ts [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+      assert.deepEqual(changeSet, { values: ['11', '12'] }, 'Removing a tag updates the filter values');
+    });
+
+    await click($('.emberTagInput-tag:contains(10)>.emberTagInput-remove')[0]);
+  });
+
+  test('error state', function(this: TestContext, assert) {
+    assert.expect(3);
+
+    assert.notOk($('.filter-values--multi-value-input--error').is(':visible'), 'The input should not have error state');
```
Comment (mattgraham1995) [inline packages/reports/tests/integration/components/filter-values/multi-value-input-test.ts:77]:
```suggestion
    assert.dom('filter-values--multi-value-input--error').isNotVisible('The input should not have error state');
```

File: packages/reports/tests/integration/components/filter-values/multi-value-input-test.ts [hunk 15]
```diff
@@ -0,0 +1,85 @@
...
+
+    this.set('filter', { validations: { isInvalid: true } });
+    assert.ok($('.filter-values--multi-value-input--error').is(':visible'), 'The input should have error state');
```
Comment (mattgraham1995) [inline packages/reports/tests/integration/components/filter-values/multi-value-input-test.ts:80]:
```suggestion
    assert.dom('.filter-values--multi-value-input--error').isVisible('The input should have error state');
```

File: packages/reports/tests/integration/components/filter-values/multi-value-input-test.ts [hunk 16]
```diff
@@ -0,0 +1,85 @@
+import { module, test } from 'qunit';
+import { setupRenderingTest } from 'ember-qunit';
+import $ from 'jquery';
```
Comment (mattgraham1995) [inline packages/reports/tests/integration/components/filter-values/multi-value-input-test.ts:3]:
```suggestion
```

**Review by mattgraham1995 — APPROVED** [review]
Just the jquery stuff, otherwise it looks good

**Pull request merged**